Resources Test Case

jorgeorpinel opened the pull request.

jorgeorpinel commented on test/Controllers/ResourcesTest.php:
```diff
@@ -0,0 +1,230 @@
+    /**
+     * Test GET /resources/things
+     * XXX Assumes there's 0 Things in the API's database.
+     * Test GonebusyLib\Controllers\ResourcesController::getResourceThings()
```
The only Q related to this PR @alexagranov, is that there's no things coming from the API so this test isn't gonna do much for now, in fact it's hardcoded to expect to receive an empty array.

alexagranov commented on test/Controllers/ResourcesTest.php:
```diff
@@ -0,0 +1,230 @@
+            case 'create':
+                return new CreateResourceBody(
+                    "name",
+                    "Staff", // Hardcoded. Way to get from API or SDK in the future?
```
Can we remove this comment?  Types will just come from API documentation.

jorgeorpinel replied:
Removed.

jorgeorpinel commented:
> For Things, I'll make a note to add some Things in Sandbox to test with in next day or two.

OK, thanks. Lmk

alexagranov commented on test/Controllers/ResourcesTest.php:
```diff
@@ -0,0 +1,233 @@
+                    "Staff",
+                    NULL, // capacity
+                    "description",
+                    NULL, // gender
```
Let's go ahead and pass a gender: `[ F | M | T/G ]`

jorgeorpinel replied:
Done. But there may be a problem with the API...

alexagranov commented:
@jorgeorpinel please resolve conflicts to continue.

jorgeorpinel commented on test/Controllers/ResourcesTest.php:
```diff
@@ -0,0 +1,234 @@
+                    GenderEnum::M, //$response->resource->gender, // XXX Hardcoded as API isn't returning a value
+                    $response->resource->name,
+                    $response->resource->thingTypeId
+                );
```
@alexagranov if you see the `XXX` codetags above, this means the API isn't returning `gender` neither after a POST /resources/new nor PUT /resources/{id}, not even on GET /resources/{id} so I had to hardcode the mocked response to match the data I sent in the first place.

![image](https://cloud.githubusercontent.com/assets/1477535/23935870/06035774-0925-11e7-8249-8904ea02e71b.png)

![image](https://cloud.githubusercontent.com/assets/1477535/23935885/1c43d13a-0925-11e7-94b5-978011c299bc.png)

alexagranov replied:
Good catch.  I'll log a bug.

jorgeorpinel commented:
@alexagranov conflicts fixed.

jorgeorpinel commented:
Good morning @alexagranov . Are you still requesting changes in this PR? BTW I removed the `XXX` comments since both problems have been logged here: API not returning gender and no Things existing in the system. For the latter, also, I changed the `getResourceThings()` test so it actually expects to find at least 3 things -- so PHPUnit will fail until the API returns them but if you review the Travis job logs there's only that one "error":
```sh
There was 1 failure:

1) ResourcesTest::testGetResourceThings
Failed asserting that actual size 0 matches expected size 3.

/home/travis/build/gonebusy/gonebusy-php-client/test/Controllers/ResourcesTest.php:207
```

jorgeorpinel commented:
p.s. Wow. It takes a while for CI to run `phpunit` now that several test cases are merged into your master... Wishing for Prism right now.

alexagranov commented:
Ok @jorgeorpinel some ThingTypes exist in Sandbox now:
```
curl -X GET --header 'Accept: application/json' --header 'Authorization: Token ac98ed08b5b0a9e7c43a233aeba841ce' 'https://sandbox.gonebusy.com/api/v1/resources/things?page=1&per_page=10'
```

Don't hate on Travis, he's a gentlemen and a scholar.  And free ;-)

jorgeorpinel commented:
Haha it's not Travis fault, I know. Just a lot of API calls to run.

So the build should pass now. How do you ask Travis to run the same job again? Hm...

jorgeorpinel commented:
@alexagranov I just updated to 0.0.9 and pushed this up so Travis ran, passing now a expected (:

alexagranov commented:
@jorgeorpinel #16 was merged 5 days ago :-)  So yes it was first.  I think I'll go ahead and merge this so we can keep moving.
